### PR TITLE
Non-experimental YAML routes with explicit schema

### PR DESCRIPTION
**Pull request by kalikiana**
- [x] Use a schema filename `JobTemplates-01.yaml` which also includes format and version
- [x] Move `job_templates_scheduling` routes out of `experimental` prefix
- [x] The `schema` needs to be specified when posting YAML
- [x] Unit tests to handle errors on missing or invalid schemas
- [ ] Publish schema under http://open.qa/api/schema/JobTemplates-01.yaml

Fixes: [poo#56426](https://progress.opensuse.org/issues/56426)

### CODE REVIEW

File: lib/OpenQA/WebAPI/Plugin/YAMLRenderer.pm [hunk 1]
```diff
@@ -44,33 +44,33 @@ sub register {
...
                 }
                 else {
-                    $schema = $validator->schema($schema_yaml);
+                    $schema = $validator->schema($schema_abspath);
                 }
             }
             catch {
-                push @errors, $_;
+                # The last line of the backtrace gives us the error message we want
+                push @errors, (split /\n/, $_)[0];
```
Comment (perlpunk) [inline lib/OpenQA/WebAPI/Plugin/YAMLRenderer.pm]: The comment says "last line", the code says first

Comment (kalikiana) [reply lib/OpenQA/WebAPI/Plugin/YAMLRenderer.pm]: D'oh. The code is right in this instance.

File: lib/OpenQA/WebAPI/Plugin/YAMLRenderer.pm [hunk 2]
```diff
@@ -44,33 +44,33 @@ sub register {
...
-            # Note: Using the schema filename; slurp'ed text isn't detected as YAML
-            my $schema_yaml = $self->app->home->child('public', 'schema', 'JobTemplate.yaml')->to_string;
-            my $validator   = JSON::Validator->new;
+            my ($self, $yaml, $schema_filename, $validate_schema) = @_;
+            my $validator = JSON::Validator->new;
             my $schema;
             my @errors;
+
             try {
+                die "No valid schema specified\n" unless ($schema_filename // '') =~ /\.yaml$/;
```
Comment (Martchus) [inline lib/OpenQA/WebAPI/Plugin/YAMLRenderer.pm]: Maybe you should adjust the regex so nobody can sneak out the designated directory using a relative path.

**Comment by kalikiana** [comment]
@okurz Any suggestions regarding publishing the schema on `open.qa`? I was looking into adding something to `generate-documentation` but held off since you said you'd suggest something for it.

**Comment by okurz** [comment]
the only suggestion I wanted to make and I did is to make you aware of help that every openQA instance provides, e.g. https://openqa.opensuse.org/help

**Comment by okurz** [comment]
If you want to publish files to open.qa then the file script/generate-documentation is the one to look into

**Comment by coolo** [comment]
open.qa is just the gh-pages branch. If the schema is not supposed to change (unless it gets a new version that is), a PR against that branch could very well do.

**Pull request merged**